src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs

```diff
@@ -140,19 +140,84 @@ public async Task<SubscriptionState> AssertSubscriptionConnectionDetails(long id
             {
                 var subscription = GetSubscriptionFromServerStore(serverStoreContext, name);
                 var topology = _serverStore.Cluster.ReadDatabaseTopology(serverStoreContext, _db.Name);
+                
                 var whoseTaskIsIt = _db.WhoseTaskIsIt(topology, subscription, subscription);
+                
                 if (whoseTaskIsIt != _serverStore.NodeTag)
                 {
+                    var databaseTopologyAvailabilityExplanation = new Dictionary<string, string>();
+
+                    string generalState = string.Empty;
+                    if (_serverStore.Engine.CurrentState == RachisState.Candidate || _serverStore.Engine.CurrentState == RachisState.Passive)
+                    {
+                        generalState = $"Current node ({_serverStore.NodeTag}) is in {_serverStore.Engine.CurrentState.ToString()} state therefore, we can't answer who's task is it and returning null";
+                    }
+                    else
+                    {
+                        generalState = _serverStore.Engine.CurrentState.ToString();
+                    }
+                    databaseTopologyAvailabilityExplanation["NodeState"] = generalState;                    
+
+                    FillNodesAvailabilityReportForState(subscription, topology, databaseTopologyAvailabilityExplanation, stateGroup:topology.Rehabs, stateName:"rehab");
+                    FillNodesAvailabilityReportForState(subscription, topology, databaseTopologyAvailabilityExplanation, stateGroup: topology.Promotables, stateName: "promotable");
+
+                    //whoseTaskIsIt!= null && whoseTaskIsIt == subscription.MentorNode 
+                    foreach (var member in topology.Members)
+                    {
+                        if (whoseTaskIsIt != null)
+                        {
+                            if (whoseTaskIsIt == subscription.MentorNode && member == subscription.MentorNode)
+                            {
+                                databaseTopologyAvailabilityExplanation[member] = "Is the mentor node and a valid member of the topology, it should be the mento node";
```

Comment:
```suggestion
                                databaseTopologyAvailabilityExplanation[member] = "Is the mentor node and a valid member of the topology, it should be the mentor node";
```

Maybe should use: 'designated' instead of mentor, though?

Reply:
Mentor is the terminology in subscriptions